app/v4/services/SubmitFinalDeclarationService.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v4.services
+
+import api.controllers.RequestContext
+import api.models
+import api.models.errors._
+import api.services.{BaseService, ServiceOutcome}
+import cats.implicits._
+import v4.connectors.SubmitFinalDeclarationConnector
+import v4.models.request.SubmitFinalDeclarationRequest
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubmitFinalDeclarationService @Inject()(connector: SubmitFinalDeclarationConnector) extends BaseService {
+
+  def submitFinalDeclaration(
+      request: SubmitFinalDeclarationRequest)(implicit ctx: RequestContext, ec: ExecutionContext): Future[ServiceOutcome[Unit]] = {
+
+    connector.submitFinalDeclaration(request).map(_.leftMap(mapDownstreamErrors(downstreamErrorMap)))
+  }
+
+  private val downstreamErrorMap: Map[String, MtdError] = {
+    val errors: Map[String, MtdError] = Map(
+      "INVALID_TAXABLE_ENTITY_ID"      -> NinoFormatError,
+      "INVALID_TAX_YEAR"               -> TaxYearFormatError,
+      "INVALID_CALCID"                 -> CalculationIdFormatError,
+      "INVALID_CORRELATION_ID"         -> models.errors.InternalError,
```

Comment:
Can we update the map without the models.error path for all the errors? 
```suggestion
      "INVALID_CORRELATION_ID"         -> InternalError,
```